ch-06/src/Solution631.java

```diff
@@ -0,0 +1,76 @@
+import java.util.Scanner;
+
+public class Solution631 {
+    public static void main(String[] args) {
+        // csak hogy megkönnyítsem a dolgod:
+        // első összeadós example: 4388576018402626, két teszt: valid: 4388576018410707, invalid: 4388576018402626
+        final Scanner input = new Scanner(System.in);
+        final long number = input.nextLong();
+
+        if (isValid(number)) {
+            System.out.println("The card number is valid");
+        } else {
+            System.out.println("The card number is invalid");
+        }
+    }
+
+    public static boolean isValid(long number) {
+        return (sumOfDoubleEvenPlace(number) + sumOfOddPlace(number)) % 10 == 0
+                && getSize(number) >= 13 && getSize(number) <= 16
+                && prefixMatched(getPrefix(number));
+    }
+
+    public static int sumOfDoubleEvenPlace(long number) {
+        int sumOfDigits = 0;
+        final String numberInString = String.valueOf(number);
+
+        for (int i = getSize(number) - 2; i >= 0; i -= 2) {
+            sumOfDigits += getDigit((numberInString.charAt(i) - (int)'0') * 2);
+        }
+        return sumOfDigits;
+    }
+
+    public static int getDigit(int number) {
+        if (number < 10) {
+            return number;
+
+        } else {
+            final String numberInString = String.valueOf(number);
+            return numberInString.charAt(0) - (int)'0' + numberInString.charAt(1) - (int)'0';
+        }
+    }
+
+    public static int sumOfOddPlace(long number) {
+        int sumOfDigits = 0;
+        final String numberInString = String.valueOf(number);
+
+        for (int i = getSize(number) - 1; i >= 0; i -= 2) {
+            sumOfDigits += numberInString.charAt(i) - (int)'0';
+        }
+        return sumOfDigits;
+    }
+
+    public static boolean prefixMatched(int d) { // a getPrefixxel tartozik össze, nemtudom a könyv mit akar, de én így oldom meg.
+        switch (d) {                                // a long number nem kell
+            case 4:
+            case 5:
+            case 6:
+            case 37: return true;
+            default: return false;
+        }
+    }

```

Comment:
The book wants you to write a `startsWith` method, just for numbers.

Thought process:

  * You need to check if the card number starts with any of these numbers:  4, 5, 37, 6.
  * For this end, you use `prefixMatched(number, d)` as `prefixMatched(cardNumber, 4) || prefixMatched(cardNumber, 5) || ...`.
  * To make this work, inside `prefixMatched` you use `getSize` to determine the size of `d` and then extract an appropriate sized prefix of `number` using `getPrefix˛`. Lastly, you compare `d` and the return value of `getPrefix`.